src/Instance.ts

```diff
@@ -105,23 +105,26 @@ export class Marked {
             }
           }
           if ('tokenizer' in ext) { // Tokenizer Extensions
-            if (!ext.level || (ext.level !== 'block' && ext.level !== 'inline')) {
+            if (ext.level && (ext.level !== 'block' && ext.level !== 'inline')) {
               throw new Error("extension level must be 'block' or 'inline'");
             }
-            const extLevel = extensions[ext.level];
-            if (extLevel) {
-              extLevel.unshift(ext.tokenizer);
-            } else {
-              extensions[ext.level] = [ext.tokenizer];
+            if (ext.position && ![...tokenizerBlockPositions, ...tokenizerInlinePositions].includes(ext.position)) {
+              throw new Error(`extension position must be one of '${tokenizerBlockPositions.join("', '")}',  '${tokenizerInlinePositions.join("', '")}'`);
+            }
+            if (!ext.level && !ext.position) {
```

Comment:
Should we enforce compatible `level` and `position` entries? I.e., a `block` level doesn't make sense with an `inline` position.  Or, should these properties be mutually-exclusive to avoid unexpected behavior?

Reply:
They are mutually exclusive. They can both be specified in case an extension wants to support many versions of marked, but versions of marked that allow position will ignore level if position is provided.